### DISCUSSION TITLE
Show node label in BoltzmannWealthModelNetwork

### DISCUSSION THREAD

**Discussion by mauruscz** [mesa/mesa#1617] [Q&A]
Hi all,

I'm working on a network-based ABM, so I'm exploiting the (incredible) functionalities of Mesa's NetworkGrid.

However, I'm struggling a bit with understanding how to interact with d3.js visualization support, probably for my lack of experience.

For example, I modified the BoltzmannWealthModelNetwork server's ```portrayal["nodes"]``` as follows:

```
    portrayal["nodes"] = [
        {
            "id": node_id,
            "size": 3, 
            "color": "#007959",
            "label": node_id,
            "showLabels": True,
            "tooltip": node_id,
        }
        for (node_id, _) in G.nodes.data("agent")
    ]
```

The tooltip functionality works (i.e. when I move the mouse over the node it shows the label), but it does not show it by default. I tried to add other parameters controlling the label size etc., but none of them seems to do the trick (neither the ```showLabels``` one is working indeed.

Can you suggest me a solution and, in general, a place where I can find some references to which are the parameters I should use?

Thank you so much for your huge work and your help,

GM

**Comment by rht** [comment]
If you want a persistent tooltip, you have to fork Mesa and modify these lines: https://github.com/projectmesa/mesa/blob/2e856679f41043b19a3376794e4c48c17c5c17a1/mesa/visualization/templates/js/NetworkModule_d3.js#L95-L104 to do what you need.
